r/R/dplyr-funcs-datetime.R

```diff
@@ -147,5 +147,12 @@ register_bindings_datetime <- function() {
   register_binding("pm", function(x) {
     !call_binding("am", x)
   })
-
+  register_binding("tz", function(x) {
+    if (call_binding("is.character", x) ||
+        call_binding("is.Date", x)) {
+      return("UTC")
```

Comment:
Hmm, I know that this is supported in lubridate, though it looks like even there it's only for backwards compatibility: 

https://github.com/tidyverse/lubridate/blob/566590f51364e6c42251cc1721f37c314ddf7e5f/R/accessors-tz.r#L21-L22

I wonder if we should add this and keep on maintaining compatibility with lubridate on a feature that they only have to maintain backwards compatibility is the right way to go... It is especially strange that something like `tz("foo")` or `tz(NA)` would return `UTC`, but even dates don't really line up to UTC (or rather: you can imagine conceptualizing a date with a timezone attached, so asserting that they are all UTC isn't quite right...)

Reply:
Yep, my initial implementation did not return `"UTC"` for strings, etc. and FWIW I think that is the way to go. I then proceeded to implement it in a `lubridate`-like manner. That allows us to decide if we want to pair it back or not. My vote would go for return timezone when timezone exists, otherwise return `NA`.

Reply:
Moreover, `lubridate` will stop returning `"UTC"` in the next update:
> Don't know how to compute timezone for object of class numeric; returning "UTC". This warning will become an error in the next major version of lubridate.

Reply:
Sounds good 👍